game/scripts/vscripts/modules/structures/modifier_fountain_aura_arena.lua

```diff
@@ -40,7 +40,7 @@ if IsServer() then
 		end
 		for i = 0, 11 do
 			local item = parent:GetItemInSlot(i)
-			if item and item:GetAbilityName() == "item_bottle_arena" then
+			if item and item:GetAbilityName() == "item_bottle_arena" and not parent:IsCourier() then
```

Comment:
What's about stash? Also, we know that parent is a courier before starting the for loop, so it may be

```lua
-- Move local hasMod ... there

if parent:IsCourier() then return end
for ...
```